### DISCUSSION TITLE
Distro - SPI hooks

### DISCUSSION THREAD

**Discussion by jamesylgan** [open-telemetry/opentelemetry-java-instrumentation#1975] [Q&A]
I am wondering - when are SPI hooks like [this](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/examples/distro/custom/src/main/resources/META-INF/services/io.opentelemetry.javaagent.spi.TracerCustomizer) required? Would you need a similar ...spi.PropagatorCustomizer file for the [DemoPropagator](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/examples/distro/custom/src/main/java/com/example/javaagent/DemoPropagator.java)?

**Comment by vovencij** [comment]
Custom propagator registration is done via TracerCustomizer, like here:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/examples/distro/custom/src/main/java/com/example/javaagent/DemoTracerCustomizer.java#L35
